cartography/intel/aws/ec2/instances.py

```diff
@@ -65,85 +65,91 @@ def transform_ec2_instances(reservations: List[Dict[str, Any]], region: str, cur
                     },
                 )
 
-            if instance.get("KeyName"):
-                key_name = instance["KeyName"]
-                key_pair_arn = f'arn:aws:ec2:{region}:{current_aws_account_id}:key-pair/{key_name}'
-                keypair_list.append({
-                    'KeyPairArn': key_pair_arn,
-                    'KeyName': key_name,
-                    'InstanceId': instance_id,
-                })
-
-            if instance.get("SecurityGroups"):
-                for group in instance["SecurityGroups"]:
-                    sg_list.append(
+                subnet_id = instance.get('SubnetId')
+                if subnet_id:
+                    subnet_list.append(
                         {
-                            'GroupId': group['GroupId'],
+                            'SubnetId': subnet_id,
                             'InstanceId': instance_id,
                         },
                     )
 
-            for network_interface in instance['NetworkInterfaces']:
-                for security_group in network_interface['Groups']:
-                    network_interface_list.append({
-                        'NetworkInterfaceId': network_interface['NetworkInterfaceId'],
-                        'Status': network_interface['Status'],
-                        'MacAddress': network_interface['MacAddress'],
-                        'Description': network_interface['Description'],
-                        'PrivateDnsName': network_interface.get('PrivateDnsName'),
-                        'PrivateIpAddress': network_interface['PrivateIpAddress'],
+                if instance.get("KeyName"):
+                    key_name = instance["KeyName"]
+                    key_pair_arn = f'arn:aws:ec2:{region}:{current_aws_account_id}:key-pair/{key_name}'
+                    keypair_list.append({
+                        'KeyPairArn': key_pair_arn,
+                        'KeyName': key_name,
                         'InstanceId': instance_id,
-                        'SubnetId': subnet_id,
-                        'GroupId': security_group['GroupId'],
                     })
 
-            if 'BlockDeviceMappings' in instance and len(instance['BlockDeviceMappings']) > 0:
-                for mapping in instance['BlockDeviceMappings']:
-                    if 'VolumeId' in mapping['Ebs']:
-                        instance_ebs_volumes_list.append({
+                if instance.get("SecurityGroups"):
+                    for group in instance["SecurityGroups"]:
+                        sg_list.append(
+                            {
+                                'GroupId': group['GroupId'],
+                                'InstanceId': instance_id,
+                            },
+                        )
+
+                for network_interface in instance['NetworkInterfaces']:
+                    for security_group in network_interface.get('Groups', []):
```

Comment:
This is the actual change that avoids the error.
The rest is just indentation after the added try/except.